### PR TITLE
Change denomination to nDusk

### PR DESCRIPTION
**Pull request by ureeves**
Changes the denomination to nDusk. This is a cross project change, but I try to centralize denomination "logic" in `rusk-abi`. If at some point we want to change this, that is where it would be done.

Resolves: #582

### CODE REVIEW

**Review by herr-seppia — APPROVED** [review]
LGTM

**Review by ZER0 — CHANGES_REQUESTED** [review]
The commits needs to have the exact same module name as prefix, so if `abi` is `rusk-abi`, `recovery` is `rusk-recovery` etc.

(Since you separated them they should be easy to be reword)

**Review by autholykos — APPROVED** [review]
LGTM

**Review by t00ts — APPROVED** [review]
GTG 👍

File: rusk-abi/src/lib.rs [hunk 1]
```diff
@@ -27,6 +28,21 @@ pub const POSEIDON_TREE_DEPTH: usize = 17;
 /// and verifier.
 pub const TRANSCRIPT_LABEL: &[u8] = b"dusk-network";
 
+/// The smallest unit of payable Dusk.
+pub type Lux = u64;
```
Comment (ZER0) [inline rusk-abi/src/lib.rs:32]: This is not exactly as discussed. I would avoid to use `Lux` at the moment. Here we'll trying to do it properly, with actual type, and then it make sense, or we just stick to `Dusk`. Basically my idea was to do not expose the internal, in this case `Lux`, as much as possible. Otherwise it doesn't even make sense using the `dusk` const fn. 

I mean, if I have something like: `dusk(1_000)` I'm implicitly thinking that what I get back is `dusk`, no matter the unit, where if I have an "additional type" it's strange that a "dusk" function returns a "lux" type: in this case, the aliasing is counter productive.

I would stick to only `Dusk` unit until we see how to do it properly with different types.

**Review by ZER0 — CHANGES_REQUESTED** [review]
Let's have another pair session if you want to, before apply the changes: it's possible I'm overlooking something.

File: rusk-abi/src/dusk.rs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+//! Dusk denomination.
+
+const DUSK_UNIT: f64 = 1_000_000_000.0;
+
+/// The minimum increment of Dusk.
+pub const LUX: Dusk = dusk(1.0 / DUSK_UNIT);
+
+/// The Dusk denomination. Use the [`dusk`] function to convert from floating
+/// point format, and the [`FromDusk`] trait to convert back.
```
Comment (ZER0) [inline rusk-abi/src/dusk.rs:15]: I believe the comment is a leftover, since we don't have the `FromDusk` trait anymore.

File: rusk-abi/src/dusk.rs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+///
+/// Values of Dusk should *never* be set directly. Instead they should always
+/// use a call to the [`dusk`] function. If increments of the smallest
+/// denomination are desired, the [`LUX`] constant can be used.
+pub type Dusk = u64;
+
+/// Convert to Dusk from floating point format.
```
Comment (ZER0) [inline rusk-abi/src/dusk.rs:22]: It should be `Converts`. Also I would rephrase as `Converts from floating point format to Dusk`, the formula `from…to` it's usually preferred, and it will be consistent with the comment below.

File: rusk-abi/src/dusk.rs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+pub const fn dusk(value: f64) -> Dusk {
+    (value * DUSK_UNIT) as Dusk
+}
+
+/// Convert from Dusk to floating point format.
```
Comment (ZER0) [inline rusk-abi/src/dusk.rs:27]: `Converts` here too (however fhe `from…to` is correct)

File: rusk-wallet/CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.4.0] - 2022-02-17
+
+## Changed
+- Use nano-dusk as the denomination [#582]
```
Comment (ZER0) [inline rusk-wallet/CHANGELOG.md:11]: Maybe this should be updated too…?

File: rusk-wallet/src/lib/mod.rs [hunk 6]
```diff
@@ -11,27 +11,12 @@ pub mod prompt;
...
-// Types that define Dusk's denomination.
-pub type Dusk = f64;
-pub type MicroDusk = u64;
+use rusk_abi::dusk::*;
 
 pub const SEED_SIZE: usize = 64;
 
-pub const ONE_MILLION: Dusk = 1_000_000.0;
+pub(crate) const MAX_CONVERTIBLE: f64 = f64::MAX / dusk(1.0) as f64;
+pub(crate) const MIN_CONVERTIBLE: f64 = from_dusk(LUX);
```
Comment (ZER0) [inline rusk-wallet/src/lib/mod.rs:19]: This should be consistent. Either we have both as:
```rust
pub(crate) const MAX_CONVERTIBLE: f64 = f64::MAX / dusk(1.0) as f64;
pub(crate) const MIN_CONVERTIBLE: f64 =  LUX as f64 / dusk(1.0) as f64;
```

or:

```rust
pub(crate) const MAX_CONVERTIBLE: f64 = from_dusk(f64::MAX);
pub(crate) const MIN_CONVERTIBLE: f64 =  from_dusk(LUX);
```

If I'm not mistaken they should be equivalent.

**Review by ZER0 — APPROVED** [review]
Looks good to me, some minor nits.
As discussed it might be not the best approach (still unsure about the `from_dusk` function) but this I believe it's the best trade off we can have at the moment that will leads to future improvements.
